### ISSUE TITLE
Allow tuples/iterables as convenience for setting contraints on model parameters

### ISSUE THREAD

**Issue by embray** [astropy/astropy#3028]
Currently it is possible to specify constraints on parameters when instantiating a model by providing keyword arguments named after the constraints, and dict that provides settings to that constraint for some (or all) parameters.  For example:

`m = SomeModel(1, 2, 3, fixed={'param_a': True, 'param_b': False})`

The dict-based assignment is definitely nice and explicit, and also good if you only need to specify some constraint on a few parameters.  But as @mhvk suggested it would also be convenient to accept a sequence that provides that constraint for all parameters.  For example if the parameters on a model are 'param_a', 'param_b', and 'param_c' one could write:

`m = SomeModel(1, 2, 3, fixed=(True, False, True))`

instead of

`m = SomeModel(1, 2, 3, fixed={'param_a': True, 'param_b': False, 'param_c': True})`

This would work since by design the parameters on each model have a standard "order" associated with them.  I see no harm in allowing both styles.

**Comment by astrofrog** [comment]
@embray, since this is marked as low priority, shall we remove the 1.0 milestone?

**Comment by Lyalpha** [comment]
Would the same behaviour be desirable for the `bound` and `tied` kwargs, or should it be restricted only to `fixed`?

**Comment by embray** [comment]
There's no reason it should be just for any one constraint, no.

**Comment by embray** [comment]
In fact at some point I plan to generalize the constraint handling code, since one thing that came out of #pyastro15 was the (eventual) need for more types of parameter constraints.

**Comment by astrofrog** [comment]
Removing milestone since not 1.1-critical

**Comment by anchitjain1234** [comment]
Please check #4574.

**Comment by anchitjain1234** [comment]
Please review #4574 .

**Comment by kakirastern** [comment]
Hi, would like to confirm that this Issue is still open?

**Comment by nden** [comment]
Yes, this issue is still open. The core of modeling is being refactored. I don't think implementing this will be in conflict with the new code but I need to confirm this with @perrygreenfield.

**Comment by kakirastern** [comment]
@nden Thanks for the reply! Will await confirmation.

**Comment by perrygreenfield** [comment]
I don't see any reason that the tuple form couldn't be supported in either version of modeling, though there are some differences in how the constraints are managed between the two, the differences are not great.

**Comment by kakirastern** [comment]
@perrygreenfield Thank you for replying! I will investigate this further and follow up soon.

**Comment by kakirastern** [comment]
Hi, I have just checked with one simple example to test whether if I could use a tuple to pass the parameters while no optional ones are present, by the following:
```
>>> import numpy as np
>>> from astropy.modeling import models

>>> g = models.Gaussian1D(amplitude=1.2, mean=0.9, stddev=0.5)
>>> h = models.Gaussian1D(1.2, 0.9, 0.5)

>>> print(g)
Model: Gaussian1D
Inputs: ('x',)
Outputs: ('y',)
Model set size: 1
Parameters:
    amplitude mean stddev
    --------- ---- ------
          1.2  0.9    0.5
>>> print(h)
Model: Gaussian1D
Inputs: ('x',)
Outputs: ('y',)
Model set size: 1
Parameters:
    amplitude mean stddev
    --------- ---- ------
          1.2  0.9    0.5

>>>  f = models.Gaussian1D(1.2, 0.9, 0.5, fixed={}, tied={}, bounds={}, eqcons=[], ineqcons=[])
>>> print(f)
Model: Gaussian1D
Inputs: ('x',)
Outputs: ('y',)
Model set size: 1
Parameters:
    amplitude mean stddev
    --------- ---- ------
          1.2  0.9    0.5
```
(astropy: 3.1.1, numpy: 1.15.4, python: 3.6.8)

Documentation for `models.Gaussian1D` can be found [here](http://docs.astropy.org/en/stable/api/astropy.modeling.functional_models.Gaussian1D.html#gaussian1d).

So that has been taken care of. But now the remaining issue seems to be with the optional parameters. At first glance, it appears that if a tuple form is to be used to pass these, then it will have to include **all or none** of the available optional parameters. _Otherwise_, this way of passing optional parameters as inputs _would not work well_. 

Any ideas regarding how to approach this, or if this should be followed up at all? Due to the amount of optional parameters available for each model, it might not really be feasible/convenient/desirable to pursue this further.

**Comment by nden** [comment]
This issue is strictly about passing the various constraints as tuples. I think the top comment describes it well. For example, I can fix `Gaussian1D.stddev` as this:

```
g=models.Gaussian1D(fixed={'amplitude':False, 'mean': False, 'stddev':True}) 

g.amplitude.fixed                                                                                                                     
Out[56]: False

g.mean.fixed                                                                                                                          
Out[57]: False

In [58]: g.stddev.fixed                                                                                                                        
Out[58]: True

```
The desired interface is
```
g = Gaussian1D(fixed=(False, False, True)
```

**Comment by kakirastern** [comment]
@nden Thanks for the correction... I guess I am still new to the wording used in the community and have misread. Will look into it a bit more. From the hints you gave me, I should be able to open a PR pretty soon.

**Comment by kakirastern** [comment]
Think I will need some kind of modification, including but not limited to the looping, ordering, and passing of constraints to be added to the file [modeling/core.py](https://github.com/astropy/astropy/blob/master/astropy/modeling/core.py) in order to attain the goal set out previously in this Issue. Will get to work on this more soon.

**Comment by kakirastern** [comment]
PR #8442 opened to address the Issue.

**Comment by LoveChauhan-18** [comment]
Hello, I would like to contribute to this issue. Would it be possible to assign it to me?
